File: cafe-api/Entities/Models/Order.cs
namespace Entities.Models
{
    public class Order
    {
        public int Id { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public decimal TotalAmount { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Processing;
        public ICollection<OrderLine> OrderLines { get; set; } = new List<OrderLine>();
        public int TableId { get; set; }
        public Table? Table { get; set; }
    }

    public enum OrderStatus
    {
        Processing,
        Delivered,
        Cancelled,
        Old
    }

}

File: cafe-web/src/types/order.ts
export default interface Order{
    id?: number;
    createdAt?: Date;
    status?: "Preparing" | "Delivered" | "Cancelled" | "Old";
    totalAmount?: number;
    orderLines: OrderLine[];
    tableId: number;
}

export interface OrderLine{
    id?: number;
    orderId?: number;
    productId: number;
    productName?: string;
    productImageUrl?: string;
    unitPrice?: number;
    quantity: number;
}

export interface OrderStatusUpdateDto {
    id: number;
    status: "Preparing" | "Delivered" | "Cancelled" | "Old";
}

export interface OrderStatsResponse {
    Hourly: OrderStats;
    Daily: OrderStats;
    Weekly: OrderStats;
    Monthly: OrderStats;
}

export interface OrderStats {
    labels: string[];
    totalCounts: number[];
    totalIncomes: number[];
}